### PR TITLE
Makes it windows compliant

### PR DESCRIPTION
**Pull request by apreg**

### CODE REVIEW

**Comment by aaronj1335** [comment]
merged to master, thanks for your help!

**Comment by aaronj1335** [comment]
oops i messed something up in the merge

**Comment by aaronj1335** [comment]
well shoot, looks like this change makes it work on windows, but not os x and linux. i had a bug in the test script that prevent this from surfacing until it was in the travis environment.

**Comment by aaronj1335** [comment]
let me do some more testing, and i may come up with something that works for everything.

**Comment by apreg** [comment]
ok

File: index.js [hunk 1]
```diff
@@ -59,28 +59,28 @@ function makeVarMap(filename) {
...
     // matches the browser's behavior
-    style.root.eachAtRule(function(atRule) {
+    style.root.eachAtRule(function (atRule) {
       if (atRule.name !== 'import')
         return;
 
       var stripped = atRule.params.replace(/^["']/, '').replace(/['"]$/, '');
 
-      process(resolveImport(stripped, dirname(filename)));
+      process(resolveImport(stripped, filename));
```
Comment (aaronj1335) [inline index.js:68]: this seems wrong -- the second argument for `resolveImport` needs to name a directory, not just any file.

**Comment by aaronj1335** [comment]
how about i push a change that fixes the unit tests, and then maybe rebase your changes on that?

**Comment by apreg** [comment]
With your changes I'm nearly 100% sure it won't work on windows. I've already tried this way but I'm open for a try.

**Comment by aaronj1335** [comment]
ok, so i committed your changes to master in 73f1aa8, realized the unit tests were failing on linux, and then reverted them in f5c3454, so yea now the master branch definitely won't work on windows.

but then i re-applied your changes with the following tweak:

``` diff
diff --git a/index.js b/index.js
index f2c5a59..ac74358 100644
--- a/index.js
+++ b/index.js
@@ -33,7 +33,7 @@ function makeVarMap(filename) {
       return path;

     if (path[0] === '.')
-      return path_.join(path_.dirname(basedir), path);
+      return path_.resolve(path_.join(basedir, path));

     // webpack treats anything starting w/ ~ as a module name, which we're
     // about to do below, so just remove leading tildes
@@ -65,7 +65,7 @@ function makeVarMap(filename) {

       var stripped = atRule.params.replace(/^["']/, '').replace(/['"]$/, '');

-      process(resolveImport(stripped, filename));
+      process(resolveImport(stripped, dirname(filename)));
     });
```

and now your changes are working for me. can you try rebasing on master, applying the diff above, and re-running the tests?

**Comment by apreg** [comment]
yeah, I can but first I have to learn some git :)

**Comment by aaronj1335** [comment]
hah yea it's complicated. let me know if i can answer questions. i'm also aaronj1335 on freenode irc if chat would be easier for you.

**Comment by aaronj1335** [comment]
actually it's a small enough change i can just apply it locally, and you're still the commit author, so that might be easiest.

you mind trying out [apregs-fix branch](https://github.com/aaronj1335/webpack-postcss-tools/tree/apregs-fix)? if the tests work for you, then i can merge that and we can close this.

**Comment by apreg** [comment]
I'm on irc in reactjs room. Can we talk there?

**Comment by apreg** [comment]
So here is what I did:

``````
git clone https://github.com/aaronj1335/webpack-postcss-tools.git into a brand new folder.
git checkout apregs-fix
npm run test
and got this error:
basedir=`dirname "$0"`
        ^
SyntaxError: Unexpected token ILLEGAL
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3```
``````

**Comment by apreg** [comment]
Dunno if you've got my message on irc but the thing is if I add webpack to package.json
and change  `if (exec('node ./node_modules/.bin/webpack --bail').code !== 0)` to 
`npm run webpack` in build.js then it works.

**Comment by aaronj1335** [comment]
apologies, i missed your irc message.

i updated the `apregs-fix` branch to use `npm run webpack` for both tests. can you give that a try?

**Comment by apreg** [comment]
No problem ;). I'm at work (I'm in UTC/GMT+1) but after work I can give it a try.

**Comment by apreg** [comment]
Good news. Everything went smooth with the latest commit in apregs-fix branch.

**Comment by aaronj1335** [comment]
rad. when i get home from work i'll merge that into master, close this out, and publish it to npm!

**Comment by aaronj1335** [comment]
merged to master and [published to npm](https://www.npmjs.com/package/webpack-postcss-tools). thanks again for all your hard work!

**Comment by apreg** [comment]
I'm glad I could help.

**Pull request closed without merging**